test/gtest/conv_embed_db.cpp

```diff
@@ -0,0 +1,196 @@
+#include <tuple>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/env.hpp>
+#include "conv_2d.hpp"
+#include "get_handle.hpp"
+
+std::string GetFloatArg()
+{
+    static const auto tmp = miopen::GetEnv("MIOPEN_TEST_FLOAT_ARG");
+    if(tmp.empty())
+    {
+        return "";
+    }
+    return tmp.front();
+};
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class ConfigWithHalf : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithInt8 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithBFloat16 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+class ConfigWithFloat : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<std::string> params;
+    switch(prec)
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenHalf: params = ConfigWithHalf::GetParam(); break;
+    case miopenInt8: params = ConfigWithInt8::GetParam(); break;
+    case miopenBFloat16: params = ConfigWithBFloat16::GetParam(); break;
+    case miopenInt8x4:
+    case miopenInt32:
+    case miopenDouble:
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "miopenInt8x4, miopenInt32, miopenDouble data type not supported by "
+                     "conv_embed_db test");
+
+    default: params = ConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        EXPECT_FALSE(capture.find("Perf Db: record not found") != std::string::npos);
+    }
+};
+
+TEST_P(ConfigWithFloat, FloatTest)
+{
+#if MIOPEN_EMBED_DB
+
+    const auto& handle = get_handle();
+    if((handle.GetDeviceName() == "gfx900" || handle.GetDeviceName() == "gfx906") &&
```

Comment:
Factor out device checks to a function.
```suggestion
    if(IsTestSupportedForDevice(handle) &&
```

Reply:
Done

Reply:
[Resolved]